### PR TITLE
Clean up the pom.xml for the connector

### LINKED ISSUE #444: Remove unused code in master branch
There are many dangling code (especially some new connector flies) in the repo that is not being used. 
It increases the barrier to fully understand things in this repo

### LINKED ISSUE #451: [Cleanup] Clean up the pom.xml for the connector
## Clean up the dependencies

- [ ] `pulsar-package-core` should be excluded.
- [ ] `flink-protobuf` should be presented in `pulsar-flink-sql-connector` with `compile` scope.
- [ ] `flink-json` is the default dependency for flink distribution. It should be optional in `pulsar-flink-sql-connector`.
- [ ] `flink-avro` should be presented in `pulsar-flink-sql-connector`.

## Clean up the maven artifact

- [ ] Remove `pulsar-flink-connector-shade-2.11`, `pulsar-flink-connector-shade-2.12`
- [ ] Remove `pulsar-flink-sql-connector-2.11`, `pulsar-flink-sql-connector-2.12`
- [ ] Create new release progress with @timmyyuan.

### PR DESCRIPTION
**Pull request by syhily**
This PR fix #444, #445 and #451. This is a huge PR, you can review it on your local IDE.

1. Cleaned up the useless dependencies.
2. Sync the checkstyle & code style from flink.
3. Format the existed code base by using a common formatter.
4. Fix some code style issues.

### CODE REVIEW

**Comment by syhily** [comment]
@nlu90 Can you help me review this clean up?

**Comment by syhily** [comment]
@timmyyuan I have changed the Github workflow for releasing `pulsar-flink`. Can you help me verify whether it's correct?

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -29,14 +29,27 @@ jobs:
         uses: actions/setup-java@v1
         with:
           java-version: 1.8
-      - name: Release Maven package
+
+      - name: Release Pulsar Flink with Scala 2.11
```
Comment (timmyyuan) [inline .github/workflows/release.yaml:33]: I think its better to use [maxtrix](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix) to summarize `scala-2.11` and `scala-2.12`. This should reduce the amount of code and increase the performance.

**Review by nlu90 — APPROVED** [review]

File: .github/workflows/release.yaml (hunk 1 above)
Comment (syhily) [reply .github/workflows/release.yaml:33]: OK, I'll fix this.

Comment (syhily) [reply .github/workflows/release.yaml]: I think that I have fixed this. Can you help me review it? @timmyyuan

**Pull request merged**